audbackend/core/backend/minio.py

```diff
@@ -0,0 +1,402 @@
+        it tries to extract missing values
+        from a config file,
+        see :meth:`get_config`.
+        If no config file exists
+        or if it has missing entries,
+        ``None`` is returned
+        for the missing entries.
+
+        Args:
+            host: hostname
+
+        Returns:
+            access token tuple
+
+        """
+        config = cls.get_config(host)
+        access_key = os.getenv("MINIO_ACCESS_KEY", config.get("access_key", None))
+        secret_key = os.getenv("MINIO_SECRET_KEY", config.get("secret_key", None))
+
+        return access_key, secret_key
+
+    @classmethod
+    def get_config(cls, host: str) -> typing.Dict:
+        """Configuration of MinIO server.
+
+        The default path of the config file is
+        :file:`~/.config/audbackend/minio.cfg`.
+        It can be overwritten with the environment variable
+        ``MINIO_CONFIG_FILE``.
+
+        If no config file can be found,
+        or no entry for the requested host,
+        an empty dictionary is returned.
+
+        The config file
+        expects one section per host,
+        e.g.
+
+        .. code-block:: ini
+
+            [play.min.io]
+            access_key = "Q3AM3UQ867SPQQA43P2F"
+            secret_key = "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG"
+
+        Args:
+            host: hostname
+
+        Returns:
+            config entries as dictionary
+
+        """
+        config_file = os.getenv("MINIO_CONFIG_FILE", "~/.config/audbackend/minio.cfg")
+        config_file = audeer.path(config_file)
+
+        if os.path.exists(config_file):
+            config = configparser.ConfigParser(allow_no_value=True)
+            config.read(config_file)
+            try:
+                config = dict(config.items(host))
+            except configparser.NoSectionError:
+                config = {}
+        else:
+            config = {}
+
+        return config
+
+    def _checksum(
+        self,
+        path: str,
+    ) -> str:
+        r"""MD5 checksum of file on backend."""
+        path = self.path(path)
+        checksum = self._client.stat_object(self.repository, path).etag
+        return checksum
+
+    def _close(
+        self,
+    ):
+        r"""Close connection to repository.
+
+        An error should be raised,
+        if the connection to the backend
+        cannot be closed.
+
+        """
+        # At the moment, this is automatically handled.
+
+    def _collapse(
+        self,
+        path,
+    ):
+        r"""Convert to virtual path.
+
+        <path>
+        ->
+        /<path>
+
+        """
+        path = "/" + path
+        path = path.replace("/", self.sep)
+        return path
+
+    def _copy_file(
+        self,
+        src_path: str,
+        dst_path: str,
+        verbose: bool,
+    ):
+        r"""Copy file on backend."""
+        src_path = self.path(src_path)
+        dst_path = self.path(dst_path)
+        # `copy_object()` has a maximum size limit of 5GB.
```

Comment:
The 5GB limit is a limit of S3 native? [This post](https://stackoverflow.com/questions/33356041/technically-what-is-the-difference-between-s3n-s3a-and-s3) suggests this at least.  So the minio client uses s3n?

Reply:
Don't know, I just found the information in the MinIO documentation. As we don't have a test for this, maybe we should also lower the value slightly to be on the safe side?

Reply:
Probably lowering would be ok to get it working right now. 
The background is a different  one: if MinIO has the limit of 5GB this suggests that the minio package  is using the old s3n interface.  Afaiu s3a succeeds s3n. The difference between is that s3n supports objects up to 5GB, but s3a should support 5TB and has improved performance. I cannof find the link right now, but I think I had seen that `fsspec` has an implementation of s3a.